src/cloudant/database.py

```diff
@@ -792,6 +793,128 @@ def shards(self):
 
         return resp.json()
 
+    def get_all_indexes(self, raw_result=False):
+        """
+        Retrieves indexes from the remote database.
+
+        :param bool raw_result: If set to True then the raw JSON content for
+            the request is returned.  Default is to return a list containing
+            :class:`~cloudant.indexes.Index`,
+            :class:`~cloudant.indexes.SearchIndex`, and
+            :class:`~cloudant.indexes.SpecialIndex` wrapped objects.
+
+        :returns: The indexes in the database
+        """
+
+        url = posixpath.join(self.database_url, '_index')
+        resp = self.r_session.get(url)
+        resp.raise_for_status()
+
+        if raw_result:
+            return resp.json()
+        
+        indexes = []
+        for data in resp.json().get('indexes', None):
```

Comment:
since you are going to iterate over this, you may want [] instead of None, since None isnt iterable